### ISSUE TITLE
How many datapoints are required for tsfresh to process an ID?

### ISSUE THREAD

**Issue by wahidelchaar** [blue-yonder/tsfresh#671]
Hi! I am trying to extract features for a classification problem. My data has about 500 ID's, each one has a time series of variable length.

This is the code that I am running to extract the features:
`feats = extract_features(data, column_id="ID", column_sort="timestamp",default_fc_parameters=params,impute_function=impute)`

After processing a dataset with 500 ID's, I am getting processed dataframes with much less ID's (varies between 0-180). I suspect this is because some of my ID's do not have enough datapoints (I confirmed that some of my ID's have 20-30 points), and so the extract_features() object is dropping these ID's.

My question is this: **What is the minimum number of datapoints required to process an ID?** I want to add this as a constraint in my SQL data preparation process (the output of which will be passed into extract_features()).

In other words, I am trying to filter out the ID's with too few points like this:
```
SELECT ID, timestamp, feature1, feature2, feature3
FROM table
GROUP BY ID
HAVING COUNT(ID) > minimum_number_of_datapoints_for_tsfresh;
```

**Comment by nils-braun** [comment]
Hi,
that is very hard to say. Which features you want to extract? And how many of them should be non-NAN? 
For example, extracting the "mean" of a time series (one of our simple features) only requires a single data point to be non-NaN.
Whereas other features might be NaN even with very long time series, because it depends on the values (and value distributions) if the time series if it is nan or not (for example some derived aggregated fft parameters might be NaN depending on the frequency distributions in the data).

Maybe as a first step, you do not impute the data on extraction and actually have a look into which features are always nan (or sometimes nan). Depending on what you want to do with the data (ML? plotting? clustering?) you could then remove those features.

**Comment by nils-braun** [comment]
@wahidelchaar Was there any follow-up question from your side? If not, I will close this issue - but please re-open if you have any more questions :-)
